### PR TITLE
Refactor the Read Namespace requests to not use the query funct…

### PR DESCRIPTION
**Pull request by mkeeler**
The query function doesn’t handle non-200 return codes properly so this ended up still trying to decode the body which resulted in weird error messages sometimes as they were not valid JSON

### CODE REVIEW

**Comment by rboyer** [comment]
Should we improve the body decoding within `query()`?

**Comment by mkeeler** [comment]
> Should we improve the body decoding within `query()`?

Maybe, the `query` function is only used in a few locations as many other read type operations already implement this same sort of logic themselves.

I didn't want to change `query` with this PR as I didn't want to potentially break existing behavior (even if that behavior is bad). This just fixes up these two functions which have yet to be included in a public release.

File: api/namespace.go [hunk 1]
```diff
@@ -100,10 +100,21 @@ func (n *Namespaces) Update(ns *Namespace, q *WriteOptions) (*Namespace, *WriteM
 
 func (n *Namespaces) Read(name string, q *QueryOptions) (*Namespace, *QueryMeta, error) {
 	var out Namespace
-	qm, err := n.c.query("/v1/namespace/"+name, &out, q)
+	r := n.c.newRequest("GET", "/v1/namespace/"+name)
+	r.setQueryOptions(q)
+	rtt, resp, err := requireOK(n.c.doRequest(r))
```
Comment (rboyer) [inline api/namespace.go]: Did you also want to handle non-200 responses, like interpret a 404 and send back a `nil` for the `*Namespace` return parameter?

Comment (mkeeler) [reply api/namespace.go]: We already will return Nil for a non-200 status code. The `requireOk` will translate a non-200 into an error and the below err check ends up returning nil.

Should we maybe return no error at all for a 404?

Comment (rboyer) [reply api/namespace.go]: Sometimes when I'm using the API i find myself wishing for more "successful errors" like having a `404` return both a `nil` item and a `nil` error. With existing methods it's tricky to fix after the fact if it changes how the method operates which is behaviorally not really backwards compatible. At least with new methods you have the option of considering the alternative.

Comment (mkeeler) [reply api/namespace.go]: I see you implemented a `requireNotFoundOrOk` for use with role management. I can switch it to use that.

**Pull request merged**